File: app/src/main/java/com/lilaclab/dadimadu/database/entity/DetailTransaksi.java
package com.lilaclab.dadimadu.database.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import com.lilaclab.dadimadu.database.entity.Produk;
import com.lilaclab.dadimadu.database.entity.Transaksi;

@Entity(tableName="detail_transaksi", primaryKeys={"id_transaksi", "id_produk"}, foreignKeys={@ForeignKey(entity=Transaksi.class, parentColumns={"id_transaksi"}, childColumns={"id_transaksi"}, onDelete=5), @ForeignKey(entity=Produk.class, parentColumns={"id_produk"}, childColumns={"id_produk"}, onDelete=1)}, indices={@Index(value={"id_transaksi"}), @Index(value={"id_produk"})})
public class DetailTransaksi {
    @NonNull
    @ColumnInfo(name="id_transaksi")
    public String idTransaksi = "";
    @ColumnInfo(name="id_produk")
    public int idProduk;
    public int qty;
    @ColumnInfo(name="harga_saat_transaksi")
    public double hargaSaatTransaksi;
    public double subtotal;
}

File: app/Models/DetailTransaksi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class DetailTransaksi extends Model
{
    protected $table      = 'detail_transaksi';
    public    $timestamps = false;
    public    $incrementing = false;

    protected $fillable = [
        'id_transaksi', 'id_produk', 'qty',
        'harga_saat_transaksi', 'subtotal',
    ];

    public function produk()
    {
        return $this->belongsTo(Produk::class, 'id_produk', 'id_produk');
    }

    public function transaksi()
    {
        return $this->belongsTo(Transaksi::class, 'id_transaksi', 'id_transaksi');
    }
}
